pages/_app.tsx

```diff
@@ -78,4 +90,27 @@ function MyApp(props: MyAppProps) {
   );
 }
 
+MyApp.getInitialProps = async ({ ctx }: { ctx: NextPageContext }) => {
+  let themeMode = '';
+  const cookiesObject: Cookies = {
+    themeTracking: '',
+    timestampTracking: '',
+    analyticsTracking: '',
+  };
+
+  if (ctx.req?.headers.cookie) {
+    const decodedCookie = decodeURIComponent(ctx?.req?.headers.cookie || '');
+    const themeModeCookie = decodedCookie?.split(';');
+    cookiesObject.themeTracking = processCookieValue(themeModeCookie[0]);
+    cookiesObject.timestampTracking = processCookieValue(themeModeCookie[1]);
+    cookiesObject.analyticsTracking = processCookieValue(themeModeCookie[2]);
+    themeMode = processCookieValue(themeModeCookie[3]);
```

Comment:
You should parse the cookies raw values and later get the values by cookies name, using the positions of the result could end up with unexpected errors as the ordering could not be the same every time, some cookies could be missing or we could add some others in the future. I recommend to use a solution like the follow one:

```js
const parseCookie = str =>
  str
  .split(';')
  .map(v => v.split('='))
  .reduce((acc, v) => {
    acc[decodeURIComponent(v[0].trim())] = decodeURIComponent(v[1].trim());
    return acc;
  }, {});
```
[source](https://www.geekstrick.com/snippets/how-to-parse-cookies-in-javascript/)

Reply:
You are right , __ will use this approach